### PR TITLE
feat: initial charm skeleton

### PR DESCRIPTION
**Pull request by yanksyoon**
Applicable spec: N/A

### Overview

This is the initial charm meta files and GitHub workflow files to break down the PR into separate chunks to allow easier PR reviews.

### Rationale

Allow easier PR reviews.
Have an initial repository setup.

### Juju Events Changes

None.

### Module Changes

None.

### Library Changes

None.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Review by yhaliaw — APPROVED** [review]

File: .github/workflows/integration_test.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+on:
+  pull_request:
+
+jobs:
+  integration-tests:
+    uses: canonical/operator-workflows/.github/workflows/integration_test.yaml@main
+    secrets: inherit
+    with:
+      juju-channel: 3.1/stable
+      provider: lxd
```
Comment (cbartz) [inline .github/workflows/integration_test.yaml:12]: what about using the edge self-hosted runners? ;-)

File: README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+For further details,
+[see the charm's detailed documentation](https://charmhub.io/tmate-ssh-server/docs).
+
+## Generating src docs for every commit
+
+Run the following command:
+
+```bash
+echo -e "tox -e src-docs\ngit add src-docs\n" > .git/hooks/pre-commit
```
Comment (cbartz) [inline README.md:43]:
```suggestion
echo -e "tox -e src-docs\ngit add src-docs\n" >> .git/hooks/pre-commit
```

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+name: Tests
+
+on:
+  pull_request:
+
+jobs:
+  unit-tests:
+    uses: canonical/operator-workflows/.github/workflows/test.yaml@debug/yanks
+    secrets: inherit
```
Comment (cbartz) [inline .github/workflows/test.yaml:9]: what about using the edge self-hosted runners? ;-)

**Review by cbartz — APPROVED** [review]
Looks good, some optional comments

File: .github/workflows/integration_test.yaml (hunk 1 above)
Comment (yanksyoon) [reply .github/workflows/integration_test.yaml:12]: Let me add them on the follow up [PR](https://github.com/canonical/tmate-ssh-server-operator/pull/5)

File: .github/workflows/test.yaml (hunk 3 above)
Comment (yanksyoon) [reply .github/workflows/test.yaml:9]: Let me add them on the follow up [PR](https://github.com/canonical/tmate-ssh-server-operator/pull/5) Thank you!

File: README.md (hunk 2 above)
Comment (yanksyoon) [reply README.md:43]: Let me add them on the follow up [PR](https://github.com/canonical/tmate-ssh-server-operator/pull/5) Thank you!

**Pull request merged**